server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -294,7 +296,9 @@ static Server makeAndInitializeServer(
       connector.setConnectionFactories(monitoredConnFactories);
     }
 
+    server.setHandler(new StatisticsHandler());
     server.setConnectors(connectors);
+    server.setAttribute(GRACEFUL_SHUTDOWN_TIMEOUT, config.getGracefulShutdownTimeout().toStandardDuration().getMillis());
 
```

Comment:
Why doesn't this call [`server.setStopTimeout`](https://www.eclipse.org/jetty/javadoc/current/org/eclipse/jetty/server/Server.html#setStopTimeout-long-) and or `server.setStopAtShutdown` which use/rely on the `StatisticsHandler` internally to cause graceful shutdown if set? As far as I can tell, using them and calling `server.stop` will have the more desirable effect of also no longer accepting new connections, which this approach seems like it will lack unless I am missing something, likely reducing the number of connections will have to be forcibly closed after the grace period is up.

further reading:
* https://github.com/eclipse/jetty.project/blob/21365234f8cf036829f7a03e5c7b89a9572444f9/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java#L204
* https://github.com/eclipse/jetty.project/blob/21365234f8cf036829f7a03e5c7b89a9572444f9/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java#L261
* https://github.com/eclipse/jetty.project/blob/385ba7d70ded912ca702ebbba2694dbcbac61256/jetty-server/src/test/java/org/eclipse/jetty/server/GracefulStopTest.java

Reply:
`setStopTimeout` sounds good, but `setStopAtShutdown` registers a shutdown hook, which should already be taken care of by the lifecycle. Shutdown hooks do not preserve any sort of order, and we want to unannounce before we shutdown jetty